src/app/services/game.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Observable, of } from 'rxjs';
+import { catchError, tap } from 'rxjs/operators';
+
+import { Game } from '../models/game';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GameService {
+  private readonly gamesUrl = `api/games`;
+  private readonly httpOptions = {
+    headers: new HttpHeaders({ 'Content-Type': 'application/json' })
+  };
+
+  constructor(private http: HttpClient) { }
+
+  addGame(game: Game): Observable<Game> {
+    return this.http.post<Game>(this.gamesUrl, game, this.httpOptions).pipe(
+      tap((newGame: Game) => console.log(`added hero w/ id=${newGame.id}`)),
+      catchError(err => {
+        console.error(`addGame failed: ${err.message}`);
+        return of(<Game>{});
+      })
+    );
+  }
+
+  getGames(): Observable<Game> {
+    return this.http.get<Game>(this.gamesUrl).pipe(
+      tap(() => console.log('fetched games')),
+      catchError(err => {
+        console.error(`getGames failed: ${err.message}`);
+        return of(<Game>{});
```

Comment:
I'm curious about the decision to return an empty Game object in the error block? If this is purely for typing purposes, here's the pattern we usually follow (using the `throwError` function provided by rxjs):

```
  getGames(): Observable<Game> { 
    return this.http.get<Game>(this.gamesUrl).pipe( 
      map((games) => { 
        console.log('fetched games'); 
        return games; 
      }), 
      catchError((err: Error) => { 
        console.error(getGames failed: ${err.message}); 
        return throwError(() => err); 
      }) 
    ); 
  }
```

Reply:
The thought was that in the event of an http error, both the game and the answer could rely on their default values so the application would continue working (albeit, you'd only be able to play Wordle with "ADEPT"). But I also see the benefit in "letting it crash" so I'm happy to update it to throw the error!

Reply:
totally up to you -- I see the value in having an offline mode, just throwing this out there in case you were returning an empty game to avoid a type error. I'm going to go ahead and approve this PR :)

Reply:
I went ahead and added `throwError`, my thought was if I do want to build out an offline mode that can be something I figure out down the line! Thank you!